### DISCUSSION TITLE
Regenerate Audio Option

### DISCUSSION THREAD

**Discussion by Dillardn7** [erew123/alltalk_tts#41] [General]
Is there a way we could get an option to regenerate the audio of a message. For example, if I were to correct the reply to properly include asterisks and quotation marks, it would be nice to be able to just regen the audio portion afterwards. While I'm at it, could there be an option to generate audio on the player side, with another drop down for player voice? (Same narrator voice) Just thoughts.

**Comment by erew123** [comment]
Both things you are asking for are theoretically possible and that's the good news! The bad news is that both are out of my hands as far as Text-gen-webui goes. 

The hamburger menu bit in text gen, next to the text entry/chat, that's all part of text-gen-webui and I have no direct control over that (I don't think I can inject any elements into it... but I will check). That would be the place to request a regen of the current text as Text-gen-webui would need to push the text out again.

Its kind of the same deal with the player voice. 

Putting it simply, Text-gen-webui spits out text at AllTalk and then AllTalk does its bits and hands back an audio file/link to Text-gen-webui. So if the main interface has an option to spit out the player text or spit out any corrected character bits, AllTalk will happily handle it. 

Technically speaking, anyone can go in and dabble in the code of Text-gen-webui, but personally Im not familiar with its code and even then it would be for Oobabooga to decide if they will or wont allow it in the code.

As I say, Ill check on the possibility of AllTalk injecting some other menu items into the hamburger menu, but I'm sceptical on it being possible. The other route of course is to put a feature request on the Text-generation-webui issues area, and see if Oobabooga or someone else who works on the interface code there may pick it up.

**Reply by Dillardn7** [reply]
Thanks! Also, thanks for writing up this extension. I'm loving it!

**Comment by erew123** [comment]
@Dillardn7 I'm closing this discussion, but I'm not closing it closing it. I'm actually putting up a central list of feature requests and I am linking back to all requests off that list. So your request will be noted in here https://github.com/erew123/alltalk_tts/discussions/74. I'm also asking on the Feature Requests list that people join in the original thread discussions by following the links.

Thanks
